toxcore/group.c

```diff
@@ -73,69 +129,47 @@ static int realloc_groupchats(Group_Chats *g_c, uint32_t num)
     return 0;
 }
 
+static void setup_conference(Group_c *g)
+{
+    memset(g, 0, sizeof(Group_c));
+    g->keep_join_index = -1;
+    g->live = true;
+}
 
 /* Create a new empty groupchat connection.
  *
  * return -1 on failure.
  * return groupnumber on success.
  */
-static int create_group_chat(Group_Chats *g_c)
+static aint create_group_chat(Group_Chats *g_c)
 {
-    uint32_t i;
+    aint i;
 
     for (i = 0; i < g_c->num_chats; ++i) {
-        if (g_c->chats[i].status == GROUPCHAT_STATUS_NONE) {
+        if (!g_c->chats[i].live) {
+            setup_conference(g_c->chats + i);
             return i;
         }
     }
 
-    int id = -1;
+    aint id = -1;
 
     if (realloc_groupchats(g_c, g_c->num_chats + 1) == 0) {
         id = g_c->num_chats;
         ++g_c->num_chats;
-        memset(&(g_c->chats[id]), 0, sizeof(Group_c));
+        setup_conference(g_c->chats + id);
     }
 
     return id;
 }
 
-
-/* Wipe a groupchat.
- *
- * return -1 on failure.
- * return 0 on success.
- */
-static int wipe_group_chat(Group_Chats *g_c, int groupnumber)
-{
-    if (groupnumber_not_valid(g_c, groupnumber)) {
-        return -1;
-    }
-
-    uint32_t i;
-    sodium_memzero(&(g_c->chats[groupnumber]), sizeof(Group_c));
-
-    for (i = g_c->num_chats; i != 0; --i) {
-        if (g_c->chats[i - 1].status != GROUPCHAT_STATUS_NONE) {
-            break;
-        }
-    }
-
-    if (g_c->num_chats != i) {
-        g_c->num_chats = i;
-        realloc_groupchats(g_c, g_c->num_chats);
-    }
-
-    return 0;
-}
-
-static Group_c *get_group_c(const Group_Chats *g_c, int groupnumber)
+static Group_c *get_group_c(const Group_Chats *g_c, aint groupnumber)
 {
-    if (groupnumber_not_valid(g_c, groupnumber)) {
-        return 0;
+    if (!is_groupnumber_valid(g_c, groupnumber)) {
+        return NULL;
     }
 
-    return &g_c->chats[groupnumber];
+    return g_c->chats + groupnumber;
```

Comment:
Why this change? I don't think the different syntax for pointer arithmetic improves clarity here.